openlibrary/plugins/upstream/addtag.py

```diff
@@ -1,27 +1,77 @@
 """Handlers for adding and editing tags."""
 
-from typing import NoReturn
-
 import web
 
 from infogami.core.db import ValidationException
 from infogami.infobase.client import ClientException
 from infogami.utils import delegate
 from infogami.utils.view import add_flash_message, public
 from openlibrary.accounts import get_current_user
-from openlibrary.plugins.upstream import spamcheck, utils
+from openlibrary.plugins.upstream import spamcheck
 from openlibrary.plugins.upstream.addbook import safe_seeother, trim_doc
 from openlibrary.plugins.upstream.models import Tag
 from openlibrary.plugins.upstream.utils import render_template
 
 
 @public
 def get_tag_types():
-    return ["subject", "work", "collection"]
+    return TAG_TYPES
+
+
+@public
+def get_subject_tag_types():
+    return SUBJECT_SUB_TYPES
+
+
+SUBJECT_SUB_TYPES = ["subject", "person", "place", "time"]
+TAG_TYPES = SUBJECT_SUB_TYPES + ["collection"]
```

Comment:
This are being used as display strings. `get_subject_tag_types` and `get_tag_types` will need to be updated such that they return lists of localized tag type strings.